crawl-ref/source/god-abil.cc

```diff
@@ -3986,6 +3987,9 @@ bool dithmenos_shadow_step()
     const actor *victim = actor_at(sdirect.target);
     mprf("You step into %s shadow.",
          apostrophise(victim->name(DESC_THE)).c_str());
+    // Using 'stepped = true' here because it's Shadow *Step*.
+    // This helps to evade splash upon landing on water.
+    moveto_location_effects(grd(old_pos), true, old_pos);
```

Comment:
oh, I had the logic backwards when we discussed this in IRC, this looks right to me.